src/lightning_fabric/wrappers.py

```diff
@@ -68,6 +71,10 @@ def step(self, closure: Optional[Callable] = None) -> Any:
             **kwargs,
         )
 
+    def zero_grad(self, **kwargs: Any) -> None:
```

Comment:
I agree with the purpose of the PR, but the implementation is different from what we have always done.

Historically, for this change, we would've rerouted all `zero_grad` calls through the strategy interface so that the `DeepSpeedStrategy` can override this hook and do this parsing.

In this case, you've skipped all that by just doing it directly in our optimizer wrapper.

Maybe this argument is more generic than just deepspeed and it makes sense to have it here. Or maybe this will eventually need to be refactored into a strategy.zero_grad().

I'm okay with this because it's a very simple validation and deepspeed themselves should fix the inconsistency, but this might go away eventually. Up to you if you want to do the full strategy change or keep it like this.

Reply:
Yes exactly, the choice here was based on your last argument. The `zero_grad` is such a simple method right now and there is currently no need for the strategy to abstract anything. If this should change in the future, we will add a public method to the strategy. Until then, Fabric and its wrappers can be that thin "frontend" translation layer.